pkg/component/kubernetes/apiserver/verticalpodautoscaler.go

```diff
@@ -51,3 +60,44 @@ func (k *kubeAPIServer) reconcileVerticalPodAutoscaler(ctx context.Context, vert
 	})
 	return err
 }
+
+func (k *kubeAPIServer) reconcileVerticalPodAutoscalerInVPAAndHPAMode(ctx context.Context, verticalPodAutoscaler *vpaautoscalingv1.VerticalPodAutoscaler, deployment *appsv1.Deployment) error {
+	updateMode := vpaautoscalingv1.UpdateModeAuto
+	kubeAPIServerMinAllowed := corev1.ResourceList{
+		corev1.ResourceCPU:    resource.MustParse("20m"),
+		corev1.ResourceMemory: resource.MustParse("200M"),
+	}
+	kubeAPIServerMaxAllowed := corev1.ResourceList{
+		// The CPU and memory are aligned to the machine ration of 1:4.
+		corev1.ResourceCPU:    resource.MustParse("7"),
+		corev1.ResourceMemory: resource.MustParse("28G"),
+	}
+
+	var evictionRequirements []*vpaautoscalingv1.EvictionRequirement
+	if k.values.Autoscaling.ScaleDownDisabled {
+		evictionRequirements = []*vpaautoscalingv1.EvictionRequirement{{
+			Resources:         []corev1.ResourceName{corev1.ResourceMemory, corev1.ResourceCPU},
+			ChangeRequirement: vpaautoscalingv1.TargetHigherThanRequests,
+		}}
+	}
```

Comment:
Nit: You could move this after L98 to avoid the declaration of the `evictionRequirements` variable (you can directly set it in the `vpa.Spec` if needed)

Reply:
No longer needed after the agreement with @voelzmo to do this via the evictions requirement controller: https://github.com/gardener/gardener/pull/9678/commits/2caae510555507424f17a2f6e6fb63cbe6d421f4